### PR TITLE
PyCBC Live creates SNR time series with slightly different sample period

### PR DESCRIPTION
**Pull request by bhooshan-gadre**
Just corrected delta_t used in followup_snr_series using round(N*delta_f).
Add assert to check delta_ts for live detectors.

### CODE REVIEW

**Comment by bhooshan-gadre** [comment]
To fix #2563

**Comment by bhooshan-gadre** [comment]
Also, round is required at_time method to get the exact trigger time. Still, trigger and series snrs are slightly different but the peak times are now same

File: bin/pycbc_live [hunk 1]
```diff
@@ -181,6 +181,9 @@ class LiveEventManager(object):
                                              data_readers, bank, followup_ifos)
 
             live_ifos = [ifo for ifo in ifos if 'snr_series' in fud[ifo]]
+            for ifo1, ifo2 in itertools.combinations(live_ifos, 2):
```
Comment (titodalcanton) [inline bin/pycbc_live]: I recommend using a one-liner with set comprehension here:
```python
assert len({fud[ifo]['snr_series'].delta_t for ifo in live_ifos}) == 1
```

File: pycbc/filter/matchedfilter.py [hunk 2]
```diff
@@ -1247,7 +1247,7 @@ def matched_filter_core(template, data, psd=None, low_frequency_cutoff=None,
         h_norm = sigmasq(htilde, psd, low_frequency_cutoff, high_frequency_cutoff)
 
     norm = (4.0 * stilde.delta_f) / sqrt( h_norm)
-    delta_t = 1.0 / (N * stilde.delta_f)
+    delta_t = 1.0 / round(N * stilde.delta_f)
```
Comment (titodalcanton) [inline pycbc/filter/matchedfilter.py]: Was this the source of the issue with the SNR series at the end? @ahnitz are you ok with this change in the matched filter core?

File: pycbc/types/timeseries.py [hunk 3]
```diff
@@ -203,7 +203,7 @@ def at_time(self, time, nearest_sample=False):
         """
         if nearest_sample:
             time += self.delta_t / 2.0
-        return self[int((time-self.start_time)*self.sample_rate)]
+        return self[int(round((time-self.start_time)*self.sample_rate))]
```
Comment (titodalcanton) [inline pycbc/types/timeseries.py]: Is this issue related to the SNR series delta_t being different?

Anyway, I think you can achieve the same result by calling `at_time` with `nearest_sample=True`, without having to do this change.

Comment (ahnitz) [inline pycbc/types/timeseries.py]: @bhooshan-gadre This should not be added. You should given the 'nearest_sample' option if you want this behavior which is already implemented.

File: pycbc/filter/matchedfilter.py (hunk 2 above)
Comment (ahnitz) [reply pycbc/filter/matchedfilter.py]: @bhooshan-gadre Would using data.delta_t work in stead?

File: pycbc/types/timeseries.py (hunk 3 above)
Comment (bhooshan-gadre) [reply pycbc/types/timeseries.py]: ah! I didn't use `nearest_sample` before assert. It is working fine.

File: pycbc/filter/matchedfilter.py (hunk 2 above)
Comment (bhooshan-gadre) [reply pycbc/filter/matchedfilter.py]: @ahnitz data.delta_t is working. I will change it so to
```suggestion
    delta_t = stilde.delta_t
```

File: bin/pycbc_live (hunk 1 above)
Comment (bhooshan-gadre) [reply bin/pycbc_live]: Yes, this willl do.
```suggestion
           assert len({fud[ifo]['snr_series'].delta_t for ifo in live_ifos}) == 1
```

**Comment by bhooshan-gadre** [comment]
Have put all the required things together in the recent commit.

**Pull request merged**